libraries/AP_Notify/Display_SSD1306_I2C.cpp

```diff
@@ -64,21 +71,25 @@ bool Display_SSD1306_I2C::hw_update()
 
     struct PACKED {
         uint8_t reg;
-        uint8_t db[SSD1306_ROWS];
+        uint8_t db[SSD1306_ROWS/2];
     } display_buffer = { 0x40, {} };
 
+
     if (!_dev || !_dev->get_semaphore()->take(5)) {
         return false;
     }
 
     // write buffer to display
     for (uint8_t i = 0; i < (SSD1306_COLUMNS / SSD1306_COLUMNS_PER_PAGE); i++) {
         command.cmd[4] = i;
-
         _dev->transfer((uint8_t *)&command, sizeof(command), nullptr, 0);
 
-        memcpy(&display_buffer.db[0], &_displaybuffer[i * SSD1306_ROWS], SSD1306_ROWS);
-        _dev->transfer((uint8_t *)&display_buffer, sizeof(display_buffer), nullptr, 0);
+        memcpy(&display_buffer.db[0], &_displaybuffer[i * SSD1306_ROWS], SSD1306_ROWS/2);
+        _dev->transfer((uint8_t *)&display_buffer, SSD1306_ROWS/2 + 1, nullptr, 0);
+
+        memcpy(&display_buffer.db[0], &_displaybuffer[i * SSD1306_ROWS + SSD1306_ROWS/2 ], SSD1306_ROWS/2);
+        _dev->transfer((uint8_t *)&display_buffer, SSD1306_ROWS/2 + 1, nullptr, 0);
```

Comment:
i split one big i2c "packet" to two parts for fixing artifacts on display when any i2c device start reset bus. as example if rgbled started but device not responding big i2c packets was broken and i see artifacts  at the screen